bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/niri/EdCompInstNIRI.java

```diff
@@ -397,4 +336,77 @@ public void textBoxKeyPress(TextBoxWidget tbwe) {
             getDataObject().setFastModeExposures(value);
         }
     }
+
+
+    /**
+     * An auxiliary method that returns a Vector with all the original NIRI Filters wrapped as Option elements,
+     * separated by Type. We use a None element to separate them in the final result. This is used for display
+     * purposes in the ComboBox
+     * @return a Vector of optional filters.
+     */
+    private static Vector<Optional<Filter>> buildElements() {
+        final java.util.List<Filter> v = new ArrayList<>(SpTypeUtil.getSelectableItems(Niri.Filter.class));
+
+        final List<Optional<Filter>> bbFilters = new ArrayList<>();
+        final List<Optional<Niri.Filter>> nbFilters = new ArrayList<>();
+
+        for (Niri.Filter filter: v) {
```

Comment:
Seems this could be done with lambdas instead as:
`v.filter(broadband::equals).foreach(f -> bbFilters.add(Optional.of(f));`
`v.filter(narrowband::equals).foreach(f -> nbFilters.add(Optional.of(f));`
to avoid the loop and switch, although this does require more linear passes through `v`, though I doubt efficiency is of a concern here.